packages/react-next/src/components/OverflowSet/OverflowSet.base.tsx

```diff
@@ -1,257 +1,225 @@
+      // Update map
+      persistedKeytips[uniqueID] = keytip;
+      delete persistedKeytips[keytipId];
     });
-  }
-
-  public render(): JSX.Element {
-    const {
-      items,
-      overflowItems,
-      className,
-      // eslint-disable-next-line deprecation/deprecation
-      focusZoneProps,
-      styles,
-      vertical,
-      // eslint-disable-next-line deprecation/deprecation
-      doNotContainWithinFocusZone,
-      role,
-      overflowSide = 'end',
-    } = this.props;
-
-    this._classNames = getClassNames(styles, { className, vertical });
-
-    let Tag;
-    let uniqueComponentProps;
-
-    if (doNotContainWithinFocusZone) {
-      Tag = 'div';
-      uniqueComponentProps = {
-        ...getNativeProps<React.HTMLAttributes<HTMLDivElement>>(this.props, divProperties),
-        ref: this._divContainer,
-      };
-    } else {
-      Tag = FocusZone;
-      uniqueComponentProps = {
-        ...getNativeProps<React.HTMLAttributes<HTMLDivElement>>(this.props, divProperties),
-        ...focusZoneProps,
-        componentRef: this._focusZone,
-        direction: vertical ? FocusZoneDirection.vertical : FocusZoneDirection.horizontal,
-      };
-    }
-
-    const showOverflow = overflowItems && overflowItems.length > 0;
-
-    return (
-      <Tag
-        role={role || 'group'}
-        aria-orientation={role === 'menubar' ? (vertical === true ? 'vertical' : 'horizontal') : undefined}
-        {...uniqueComponentProps}
-        className={this._classNames.root}
-      >
-        {overflowSide === 'start' && showOverflow && this._onRenderOverflowButtonWrapper(overflowItems!)}
-        {items && this._onRenderItems(items)}
-        {overflowSide === 'end' && showOverflow && this._onRenderOverflowButtonWrapper(overflowItems!)}
-      </Tag>
-    );
-  }
-
-  /**
-   * Sets focus to the first tabbable item in the OverflowSet.
-   * @param forceIntoFirstElement - If true, focus will be forced into the first element,
-   * even if focus is already in theOverflowSet
-   * @returns True if focus could be set to an active element, false if no operation was taken.
-   */
-  public focus(forceIntoFirstElement?: boolean): boolean {
-    let focusSucceeded = false;
-
+    return () => {
+      // Delete all persisted keytips saved
+      Object.keys(persistedKeytips).forEach((uniqueID: string) => {
+        keytipManager.unregister(persistedKeytips[uniqueID] as IKeytipProps, uniqueID, true);
+      });
+      persistedKeytips = {};
+    };
+  }, [persistedKeytips, keytipManager]);
+};
+
+const useComponentRef = (
+  props: IOverflowSetProps,
+  focusZone: React.RefObject<IFocusZone>,
+  divContainer: React.RefObject<HTMLDivElement>,
+) => {
+  React.useImperativeHandle(
+    props.componentRef,
+    (): IOverflowSet => ({
+      focus: (forceIntoFirstElement?: boolean): boolean => {
+        let focusSucceeded = false;
+        // eslint-disable-next-line deprecation/deprecation
+        if (props.doNotContainWithinFocusZone) {
```

Comment:
We have a note in the `focusZoneProps` and `doNotContainWithinFocusZone` docs which says "In 8.0 the OverflowSet will no longer be wrapped in a FocusZone" (added by @micahgodbolt). I think we should go ahead and make that change now.

Removing these props makes the original comment I was going to leave here irrelevant, but the general issue is something to be careful about elsewhere: since this imperative handle has no dependencies, references to `props` will only ever refer to the *first* props passed in.